src/plugins/intel_cpu/src/nodes/scatter_update.cpp

```diff
@@ -263,6 +263,377 @@ static std::vector<size_t> getBlockND(const VectorDims& shape) {
+                ptr_dst_offset = &dst_offsets[0];
+                for (size_t worker = start; worker < end; worker++) {
+                    auto indices_offset = *ptr_indices_offset + idx * indicesBlockND[axis + 1];
+                    IndexType idxValue = *(indicesPtr + indices_offset);
+                    if (idxValue < 0) idxValue += data_dim_size;
+                    ASSERT_DEBUG_ONLY(idxValue < data_dim_size && idxValue >= 0, "invalid index value.");
+                    auto dst = dataPtr + (*ptr_dst_offset + idxValue * dataBlockND[axis + 1]);
+                    auto src = updatePtr + indices_offset;
+                    kernel_func(dst, src);
+                    ptr_indices_offset++;
+                    ptr_dst_offset++;
+                }
+            }
+        }
+    });
+}
+
+template <typename DataType, typename IndexType>
+void scatterElementsUpdate(const MemoryPtr& mem_data, const MemoryPtr& mem_indices, const MemoryPtr& mem_updates,
+                            int axis, const ScatterUpdate::Config& config, ReduceMean& kernel_func) {
+    PlainTensor data_buf, indices_buf, updates_buf;
+    data_buf.reset(mem_data);
+    indices_buf.reset(mem_indices);
+    updates_buf.reset(mem_updates);
+
+    const auto& data_shape = mem_data->getStaticDims();
+    const auto& indices_shape = mem_indices->getStaticDims();
+    size_t updates_rank = indices_shape.size();
+
+    const int64_t data_dim_size = static_cast<int64_t>(data_shape[axis]);
+    const auto index_dim_size = indices_shape[axis];
+
+    const bool use_init_val = config.use_init_val;
+    const Reduction reduction_type = config.reduction_type;
+
+    if (axis < 0)
+        axis += updates_rank;
+
+    VectorDims squashed_indices_shape(indices_shape);
+    squashed_indices_shape[axis] = 1;
+
+    // process serially along 'axis' dimension because of data dependency brought by duplicated value in indices
+    parallel_nt(0, [&](const int ithr, const int nthr) {
+        size_t start = 0, end = 0;
+        splitter(shape_size(squashed_indices_shape), nthr, ithr, start, end);
+
+        if (!use_init_val) {
+            const auto value = reduction_neutral_value<DataType>(reduction_type);
+            for (size_t worker = start; worker < end; worker++) {
+                VectorDims indices_coord(updates_rank, 0);
+                getCoordinate(indices_coord, worker, squashed_indices_shape);
+                std::vector<size_t> data_coord(indices_coord);
+
+                for (size_t i = 0; i < index_dim_size; i++) {
+                    indices_coord[axis] = i;
+                    IndexType idxValue = indices_buf.at<IndexType, size_t>(indices_coord);
+                    if (idxValue < 0) idxValue += data_dim_size;
+                    ASSERT_DEBUG_ONLY(idxValue < data_dim_size && idxValue >= 0, "invalid index value.");
+                    data_coord[axis] = idxValue;
+                    data_buf.at<DataType, size_t>(data_coord) = value;
```

Comment:
why use different way to iterate the tensor in this specialization ? can we use `TensorIterator` in both generic & reduce cases?

Reply:
refacted